npm/webpack-preprocessor/test/unit/cross-origin-callback-loader.spec.ts

```diff
@@ -0,0 +1,213 @@
+          })
+        })`,
+      )
+
+      expectAddFileSource(store).to.equal(stripIndent`
+        __cypressCrossOriginCallback = function () {
+          require('../support/utils');
+        }`)
+    })
+
+    it('works when dep is not assigned to a variable', () => {
+      const { store } = callLoader(
+        `it('test', () => {
+          cy.origin('http://foobar.com:3500', () => {
+            Cypress.require('../support/utils')
+          })
+        })`,
+      )
+
+      expectAddFileSource(store).to.equal(stripIndent`
+        __cypressCrossOriginCallback = () => {
+          require('../support/utils');
+        }`)
+    })
+
+    it('works when dep is assigned to a variable', () => {
+      const { store } = callLoader(
+        `it('test', () => {
+          cy.origin('http://foobar.com:3500', () => {
+            const utils = Cypress.require('../support/utils')
+            utils.foo()
+          })
+        })`,
+      )
+
+      expectAddFileSource(store).to.equal(stripIndent`
+        __cypressCrossOriginCallback = () => {
+          const utils = require('../support/utils');
+
+          utils.foo();
+        }`)
+    })
+
+    it('works with multiple Cypress.require()s', () => {
+      const { store } = callLoader(
+        `it('test', () => {
+          cy.origin('http://foobar.com:3500', () => {
+            Cypress.require('../support/commands')
+            const utils = Cypress.require('../support/utils')
+            const _ = Cypress.require('lodash')
+          })
+        })`,
+      )
+
+      expectAddFileSource(store).to.equal(stripIndent`
+        __cypressCrossOriginCallback = () => {
+          require('../support/commands');
+
+          const utils = require('../support/utils');
+
+          const _ = require('lodash');
+        }`)
+    })
+
+    it('works when .origin() is chained off another command', () => {
+      const { store } = callLoader(
+        `it('test', () => {
+          cy
+          .wrap({})
+          .origin('http://foobar.com:3500', () => {
+            Cypress.require('../support/commands')
+          })
+        })`,
+      )
+
+      expectAddFileSource(store).to.equal(stripIndent`
+        __cypressCrossOriginCallback = () => {
+          require('../support/commands');
+        }`)
+    })
+
+    it('works when result of require() is invoked', () => {
+      const { store } = callLoader(
+        `it('test', () => {
+          cy.origin('http://foobar.com:3500', () => {
+            const someVar = 'someValue'
+            const result = Cypress.require('./fn')(someVar)
+            expect(result).to.equal('mutated someVar')
+          })
+        })`,
+      )
+
+      expectAddFileSource(store).to.equal(stripIndent`
+        __cypressCrossOriginCallback = () => {
+          const someVar = 'someValue';
+
+          const result = require('./fn')(someVar);
+
+          expect(result).to.equal('mutated someVar');
+        }`)
+    })
```

Comment:
Added that test in https://github.com/cypress-io/cypress/pull/23283/commits/86225f2367a95f85dcc3b6dab017c8c96794e099

Reply:
The example

```js
cy.origin('http://foobar.com:3500', { args: undefined}}, (foo = Cypress.require('foo')) => {})
```

isn't valid because the `Cypress.require()` isn't inside the callback itself.

A different scenario that's not currently handled is assigning the callback to a variable and passing that into `cy.origin()`:

```js
const callback = () => {
  Cypress.require('foo')
}

cy.origin('http://foobar.com:3500', callback)
```

It _might_ be possible to replace it in that case, but I think that should be a later enhancement.

Reply:
I think that is completely reasonable. Especially if a user is doing something like that the likelihood of a workaround is extremely high.